### PR TITLE
[e2e] bump test-infra with better docker support

### PR DESCRIPTION
**Pull request by pducolin**
### What does this PR do?

Integrate changes from https://github.com/DataDog/test-infra-definitions/pull/501

### Motivation

Some time ago @nplanel asked us to allow accessing the VM in a docker scenario: when a docker compose shares a folder with the host, we want to allow operating on the host.

At the same time we received few requests to support fakeintake in a docker scenario.

To allow all of the above, I moved the VM creation out of the Daemon. We can now call

daemon, err := Daemon(vm, options...)
to install a dockerized Agent on a UnixVm with docker installed. This is closer to what we do to install the agent on a VM

installer, err := agent.NewInstaller(vm, agentParameters...)
This will not change the generic usage in tests datadog-agent: the implementation change will still be wrapped inside a `DockerStackDef`.

### Additional Notes

Changes: https://github.com/DataDog/test-infra-definitions/compare/aa1982fa029c...f422c1f4dfae

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: test/new-e2e/test-infra-definition/docker_test.go [hunk 1]
```diff
@@ -9,18 +9,21 @@ import (
...
+	"github.com/stretchr/testify/assert"
 )
 
 type dockerSuite struct {
 	e2e.Suite[e2e.DockerEnv]
 }
 
 func TestDocker(t *testing.T) {
-	e2e.Run(t, &dockerSuite{}, e2e.DockerStackDef(dockerparams.WithAgent()))
+	e2e.Run(t, &dockerSuite{}, e2e.DockerStackDef(os.AMD64Arch), params.WithDevMode())
```
Comment (KevinFairise2) [inline test/new-e2e/test-infra-definition/docker_test.go:26]: We should not keep dev mode here, right?

**Review by KevinFairise2 — COMMENTED** [review]
Small comment otherwise LGTM!

File: test/new-e2e/test-infra-definition/docker_test.go (hunk 1 above)
Comment (pducolin) [reply test/new-e2e/test-infra-definition/docker_test.go:26]: Nice catch ! 

`WithDevMode` is disabled on the CI, but I didn't mean to keep this in code !

**Review by KevinFairise2 — APPROVED** [review]

**Review by AlexandreYang — APPROVED** [review]
good for ndm 👍

**Review by dplepage-dd — APPROVED** [review]
LGTM for ndm

**Pull request merged**